hail/src/main/scala/is/hail/io/BufferSpecs.scala

```diff
@@ -99,23 +130,50 @@ trait BlockBufferSpec extends Spec {
   def nativeInputBufferType(inputStreamType: String): String
 }
 
-final case class LZ4BlockBufferSpec(blockSize: Int, child: BlockBufferSpec) extends BlockBufferSpec {
+sealed abstract class LZ4BlockBufferSpecCommon extends BlockBufferSpec {
   require(blockSize <= (1 << 16))
 
-  def buildInputBuffer(in: InputStream): InputBlockBuffer = new LZ4InputBlockBuffer(blockSize, child.buildInputBuffer(in))
+  def typeName: String
+
+  def lz4: LZ4
 
-  def buildOutputBuffer(out: OutputStream): OutputBlockBuffer = new LZ4OutputBlockBuffer(blockSize, child.buildOutputBuffer(out))
+  def blockSize: Int
+
+  def child: BlockBufferSpec
+
+  def buildInputBuffer(in: InputStream): InputBlockBuffer = new LZ4InputBlockBuffer(lz4, blockSize, child.buildInputBuffer(in))
+
+  def buildOutputBuffer(out: OutputStream): OutputBlockBuffer = new LZ4OutputBlockBuffer(lz4, blockSize, child.buildOutputBuffer(out))
 
   def buildCodeInputBuffer(in: Code[InputStream]): Code[InputBlockBuffer] =
     Code.newInstance[LZ4InputBlockBuffer, Int, InputBlockBuffer](blockSize, child.buildCodeInputBuffer(in))
 
   def buildCodeOutputBuffer(out: Code[OutputStream]): Code[OutputBlockBuffer] =
     Code.newInstance[LZ4OutputBlockBuffer, Int, OutputBlockBuffer](blockSize, child.buildCodeOutputBuffer(out))
 
-  def nativeOutputBufferType: String = s"LZ4OutputBlockBuffer<${ 4 + LZ4Utils.maxCompressedLength(blockSize) }, ${ child.nativeOutputBufferType }>"
+  def nativeOutputBufferType: String = s"$typeName<${ 4 + lz4.maxCompressedLength(blockSize) }, ${ child.nativeOutputBufferType }>"
```

Comment:
this method doesn't even seem to be used. Can we delete?

Reply:
I will give it a try